### PR TITLE
Proposal cards: App badges

### PR DESCRIPTION
**Pull request by delfipolito**
image

### CODE REVIEW

File: src/hooks/useDisputableVotes.js [hunk 1]
```diff
@@ -133,16 +136,40 @@ async function processVote(vote, apps) {
...
     status: vote.status,
+    target: getTargetData(apps, disputableVotingApp.address),
   }
 
   if (vote.script === EMPTY_SCRIPT) {
     return extendedVote
   }
 
   const description = await describeScript(vote.script, apps)
+  if (description[0] && description[0].to) {
```
Comment (andy-hook) [inline src/hooks/useDisputableVotes.js:147]: Should we destructure / assign this on a previous line to save us accessing via the index each time?

File: src/hooks/useDisputableVotes.js [hunk 2]
```diff
@@ -133,16 +136,40 @@ async function processVote(vote, apps) {
...
+      icon: '',
+    }
+  }
+
+  if (
+    apps &&
+    apps.filter((app) => app.address === extendedVote.target.address).length > 0
+  ) {
+    extendedVote.target = getTargetData(apps, extendedVote.target.address)
+  }
```
Comment (andy-hook) [inline src/hooks/useDisputableVotes.js:160]: Find this logic a little bit confusing, perhaps we could clarify the intention by assigning these as descriptive variables / adding comments? Eventually we might want to break them up into composable functions but in the interest of time just a little bit more description here would be useful.

**Review by andy-hook — COMMENTED** [review]
Can't wait to get this in the app 😍

I've added a few small things, the logic of getting the appearance is a little confusing by the nature of the checks. I think making it more descriptive would be good in the short term so we have it ready for the PC, but we will probably want to re-think how it's structured as to make it clearer for the future.

File: src/hooks/useDisputableVotes.js (hunk 2 above)
Comment (delfipolito) [reply src/hooks/useDisputableVotes.js]: i tried to explain this a little bit more. let me know if this is ok :)

File: src/components/Proposals/ProposalCard.js [hunk 3]
```diff
@@ -69,7 +69,14 @@ function ProposalCard({ appAddress, vote, onProposalClick }) {
...
       >
-        <AppBadge label="Disputable Voting" appAddress={appAddress} badgeOnly />
+        {target && (
+          <AppBadge
+            label={target.name ? target.name : target.address}
+            appAddress={target.address}
+            iconSrc={target.icon ? target.icon : ''}
+            badgeOnly
+          />
+        )}
```
[the remark this answers is not in the document]
Comment (delfipolito) [reply src/components/Proposals/ProposalCard.js]: is not necessary anymore

File: src/hooks/useDisputableVotes.js [hunk 4]
```diff
@@ -133,16 +136,46 @@ async function processVote(vote, apps) {
     naysPct: vote.naysPct,
     yeasPct: vote.yeasPct,
     status: vote.status,
+    target: getTargetData(apps, disputableVotingApp.address),
   }
 
+  // return with disputableVotingApp as the default app target
```
Comment (andy-hook) [inline src/hooks/useDisputableVotes.js:142]:
```suggestion
  // Return disputableVotingApp as the default app target
```

File: src/hooks/useDisputableVotes.js [hunk 5]
```diff
@@ -133,16 +136,46 @@ async function processVote(vote, apps) {
...
   if (vote.script === EMPTY_SCRIPT) {
     return extendedVote
   }
 
   const description = await describeScript(vote.script, apps)
+  const voteHasAppTarget = description[0] && description[0].to
+
+  // if we have the target app on the description, we change the default one
```
Comment (andy-hook) [inline src/hooks/useDisputableVotes.js:150]:
```suggestion
  // If we have the target app on the description, build the target data with it instead
```

File: src/hooks/useDisputableVotes.js [hunk 6]
```diff
@@ -133,16 +136,46 @@ async function processVote(vote, apps) {
...
+  if (voteHasAppTarget) {
+    extendedVote.target = {
+      address: description[0].to,
+      name: description[0].name || description[0].identifier,
+    }
+  }
+
+  // if the target app is one of the known ones
+  // we try to get the icon and name from there
```
Comment (andy-hook) [inline src/hooks/useDisputableVotes.js:159]:
```suggestion
  // If the target app is known to us then we can get it's icon and name 
```

**Review by andy-hook — APPROVED** [review]
Couple of amends to the comments, are these accurate?

**Pull request merged**